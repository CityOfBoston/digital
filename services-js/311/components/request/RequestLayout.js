// @flow

import React from 'react';
import { css } from 'glamor';
import Router from 'next/router';
import type { Context } from 'next';
import { observer } from 'mobx-react';

import type { RequestAdditions } from '../../server/next-handlers';

import Nav from '../common/Nav';
import { LocationMapWithLibrary } from '../map/LocationMap';
import type { MapMode } from '../map/LocationMap';
import HomeDialog from './home/HomeDialog';
import type { InitialProps as HomeDialogInitialProps } from './home/HomeDialog';
import RequestDialog from './request/RequestDialog';
import type { InitialProps as RequestDialogInitialProps } from './request/RequestDialog';
import TranslateDialog from './translate/TranslateDialog';

import { HEADER_HEIGHT } from '../style-constants';

import makeLoopbackGraphql from '../../data/dao/loopback-graphql';
import type { LoopbackGraphql } from '../../data/dao/loopback-graphql';

import type { AppStore } from '../../data/store';

type HomeData = {|
  view: 'home',
  props: HomeDialogInitialProps,
|};

type RequestData = {|
  view: 'request',
  props: RequestDialogInitialProps,
|};

type TranslateData = {|
  view: 'translate',
  props: {},
|};

export type InitialProps = {|
  data: HomeData | RequestData | TranslateData,
|};

export type Props = {|
  store: AppStore,
  noMap?: boolean,
  ...InitialProps,
|};

type State = {|
  locationMapActive: boolean,
|};

// This is the main container for content. We want to be at least full height on
// large screens to push the footer down to where you need to scroll for it.
const CONTAINER_STYLE = css({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  position: 'relative',
});

const BACKGROUND_MAP_CONTAINER_STYLE = css({
  position: 'fixed',
  width: '100%',
  top: HEADER_HEIGHT,
  bottom: 0,
  background: '#9B9B9B',
});

// We have one class for picking the service type and doing the entire request
// so that we can keep a consistent Map background behind the
// dialog and not repeatedly re-mount it.
//
// This class has two "views": the "summaries" view that lets the user pick a
// service request, and the "service" view for the request flow. That flow
// proceeds through the "questions" "location" and "contact" stages.
@observer
export default class RequestLayout extends React.Component {
  static defaultProps = {
    noMap: false,
  };

  props: Props;
  state: State = {
    locationMapActive: false,
  };
  loopbackGraphql: LoopbackGraphql = makeLoopbackGraphql();

  static async getInitialProps(
    ctx: Context<RequestAdditions>
  ): Promise<InitialProps> {
    const { query } = ctx;

    let data;

    if (query.translate === '1') {
      data = {
        view: 'translate',
        props: {},
      };
    } else if (query.code) {
      data = {
        view: 'request',
        props: await RequestDialog.getInitialProps(ctx),
      };
    } else {
      data = {
        view: 'home',
        props: await HomeDialog.getInitialProps(ctx),
      };
    }

    return {
      data,
    };
  }

  componentDidMount() {
    const { store } = this.props;
    store.mapLocation.start(this.loopbackGraphql);
  }

  componentWillReceiveProps() {
    this.setState({ locationMapActive: false });
  }

  componentWillUnmount() {
    const { store } = this.props;
    store.mapLocation.stop();
  }

  routeToServiceForm = async (code: string, stage: string = 'questions') => {
    if (stage === 'questions') {
      await Router.push(`/request?code=${code}`, `/request/${code}`);
    } else {
      await Router.push(
        `/request?code=${code}&stage=${stage}`,
        `/request/${code}/${stage}`
      );
    }

    window.scrollTo(0, 0);
  };

  setLocationMapActive = (active: boolean) => {
    const { locationMapActive } = this.state;

    if (locationMapActive !== active) {
      this.setState({ locationMapActive: active });
    }
  };

  render() {
    const { data, store, noMap } = this.props;
    const { locationMapActive } = this.state;
    const { ui: { mediaLarge } } = store;

    let mapMode: MapMode;
    if (locationMapActive) {
      mapMode = 'picker';
    } else {
      mapMode = 'inactive';
    }

    return (
      <div>
        <Nav activeSection="request" />

        {/* Outer box needs to take up at least the screen size on desktop, so
            that the content can center over the map and keep the footer from
            encroaching up. */}
        <div
          className={CONTAINER_STYLE}
          style={{ backgroundColor: 'transparent' }}
          role="main"
        >
          {(mediaLarge || !process.browser) &&
            !noMap &&
            <div className={BACKGROUND_MAP_CONTAINER_STYLE}>
              <LocationMapWithLibrary
                store={store}
                mode={mapMode}
                mobile={false}
              />
            </div>}

          {/* We can't put any wrappers around the dialogs below because in the
               case of RequestDialog's location picker, it needs to make sure nothing
               is overlapping the map so it can get clicks. */}

          {data.view === 'home' &&
            <HomeDialog
              store={store}
              loopbackGraphql={this.loopbackGraphql}
              {...data.props}
            />}

          {data.view === 'request' &&
            <RequestDialog
              store={store}
              loopbackGraphql={this.loopbackGraphql}
              routeToServiceForm={this.routeToServiceForm}
              setLocationMapActive={this.setLocationMapActive}
              {...data.props}
            />}

          {data.view === 'translate' &&
            <TranslateDialog languages={store.languages} />}
        </div>
      </div>
    );
  }
}
