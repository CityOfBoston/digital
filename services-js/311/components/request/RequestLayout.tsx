import React from 'react';
import { css } from 'emotion';
import Router from 'next/router';
import { observer } from 'mobx-react';

import { NextContext } from '@cityofboston/next-client-common';
import { HEADER_HEIGHT } from '@cityofboston/react-fleet';

import FeedbackBanner from '../common/FeedbackBanner';
import Footer from '../common/Footer';
import Nav from '../common/Nav';
import { LocationMapWithLibrary } from '../map/LocationMap';
import { MapMode } from '../map/LocationMap';
import HomeDialog from './home/HomeDialog';
import { InitialProps as HomeDialogInitialProps } from './home/HomeDialog';
import RequestDialog from './request/RequestDialog';
import { InitialProps as RequestDialogInitialProps } from './request/RequestDialog';
import TranslateDialog from './translate/TranslateDialog';

import {
  PageDependencies,
  GetInitialPropsDependencies,
} from '../../pages/_app';

type HomeData = {
  view: 'home';
  props: HomeDialogInitialProps;
};

type RequestData = {
  view: 'request';
  props: RequestDialogInitialProps;
};

type TranslateData = {
  view: 'translate';
  props: {};
};

export type InitialProps = {
  data: HomeData | RequestData | TranslateData;
};

export type Props = Pick<
  PageDependencies,
  // Some of these we need just for LocationMap
  | 'addressSearch'
  | 'fetchGraphql'
  | 'ui'
  | 'requestSearch'
  | 'browserLocation'
  | 'siteAnalytics'
  | 'liveAgent'
  | 'screenReaderSupport'
  | 'languages'
> &
  InitialProps & {
    noMap?: boolean;
  };

type State = {
  locationMapActive: boolean;
};

// This is the main container for content. We want to be at least full height on
// large screens to push the footer down to where you need to scroll for it.
const CONTAINER_STYLE = css({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  position: 'relative',
});

const BACKGROUND_MAP_CONTAINER_STYLE = css({
  position: 'fixed',
  width: '100%',
  top: HEADER_HEIGHT,
  bottom: 0,
  background: '#9B9B9B',
});

// We have one class for picking the service type and doing the entire request
// so that we can keep a consistent Map background behind the
// dialog and not repeatedly re-mount it.
//
// This class has two "views": the "summaries" view that lets the user pick a
// service request, and the "service" view for the request flow. That flow
// proceeds through the "questions" "location" and "contact" stages.
@observer
export default class RequestLayout extends React.Component<Props, State> {
  static defaultProps = {
    noMap: false,
  };

  state: State = {
    locationMapActive: false,
  };

  static async getInitialProps(
    ctx: NextContext,
    deps: Pick<GetInitialPropsDependencies, 'fetchGraphql'>
  ): Promise<InitialProps> {
    const { query } = ctx;

    let data;

    if (query.translate === '1') {
      data = {
        view: 'translate',
        props: {},
      };
    } else if (query.code) {
      data = {
        view: 'request',
        props: await RequestDialog.getInitialProps(ctx, deps),
      };
    } else {
      data = {
        view: 'home',
        props: await HomeDialog.getInitialProps(ctx, deps),
      };
    }

    return {
      data,
    };
  }

  componentDidMount() {
    const { addressSearch, fetchGraphql } = this.props;
    addressSearch.start(fetchGraphql);
  }

  componentWillReceiveProps() {
    this.setState({ locationMapActive: false });
  }

  componentWillUnmount() {
    const { addressSearch } = this.props;
    addressSearch.stop();
  }

  routeToServiceForm = async (code: string, stage: string = 'questions') => {
    if (stage === 'questions') {
      await Router.push(`/request?code=${code}`, `/request/${code}`);
    } else {
      await Router.push(
        `/request?code=${code}&stage=${stage}`,
        `/request/${code}/${stage}`
      );
    }

    window.scrollTo(0, 0);
  };

  setLocationMapActive = (active: boolean) => {
    const { locationMapActive } = this.state;

    if (locationMapActive !== active) {
      this.setState({ locationMapActive: active });
    }
  };

  render() {
    const {
      data,
      ui,
      noMap,
      requestSearch,
      addressSearch,
      browserLocation,
      siteAnalytics,
      liveAgent,
      fetchGraphql,
      screenReaderSupport,
      languages,
    } = this.props;
    const { locationMapActive } = this.state;
    const { mediaLarge } = ui;

    let mapMode: MapMode;
    if (locationMapActive) {
      mapMode = 'picker';
    } else {
      mapMode = 'inactive';
    }

    return (
      <div>
        <Nav activeSection="request" />

        {/* Outer box needs to take up at least the screen size on desktop, so
            that the content can center over the map and keep the footer from
            encroaching up. */}
        <div
          className={`mn--full ${CONTAINER_STYLE.toString()}`}
          style={{ backgroundColor: 'transparent' }}
          role="main"
        >
          <div className={`mn-fixed-child ${BACKGROUND_MAP_CONTAINER_STYLE}`}>
            {/* Condition must be on the inner element. If it's on the outer
            DIV, React 16's reconcilliation of the server and client (which are
            different because we're conditional on mediaLarge) will apply the
            runtime attributes to the server content. And things will be broken.*/}
            {(mediaLarge || !process.browser) && !noMap && (
              <LocationMapWithLibrary
                addressSearch={addressSearch}
                browserLocation={browserLocation}
                requestSearch={requestSearch}
                ui={ui}
                mode={mapMode}
                mobile={false}
              />
            )}
          </div>

          {/* We can't put any wrappers around the dialogs below because in the
               case of RequestDialog's location picker, it needs to make sure nothing
               is overlapping the map so it can get clicks. */}

          {data.view === 'home' && (
            <HomeDialog
              fetchGraphql={fetchGraphql}
              siteAnalytics={siteAnalytics}
              ui={ui}
              liveAgent={liveAgent}
              languages={languages}
              bypassTranslateDialog={data.props.bypassTranslateDialog}
              description={data.props.description}
              stage={data.props.stage}
              topServiceSummaries={data.props.topServiceSummaries}
            />
          )}

          {data.view === 'request' && (
            <RequestDialog
              fetchGraphql={fetchGraphql}
              addressSearch={addressSearch}
              browserLocation={browserLocation}
              requestSearch={requestSearch}
              screenReaderSupport={screenReaderSupport}
              ui={ui}
              routeToServiceForm={this.routeToServiceForm}
              setLocationMapActive={this.setLocationMapActive}
              description={data.props.description}
              service={data.props.service}
              serviceCode={data.props.serviceCode}
              stage={data.props.stage}
            />
          )}

          {data.view === 'translate' && (
            <TranslateDialog languages={languages} />
          )}
        </div>

        <Footer />
        <FeedbackBanner fit="DIALOG" />
      </div>
    );
  }
}
