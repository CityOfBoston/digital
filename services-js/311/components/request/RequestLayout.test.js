// @flow
/* eslint no-fallthrough: 0 */

import * as React from 'react';
import renderer from 'react-test-renderer';
import Router from 'next/router';

import type { Service, ServiceSummary } from '../../data/types';

import { makeServerContext } from '../../lib/test/make-context';

import getStore from '../../data/store';

import RequestLayout from './RequestLayout';

jest.mock('next/router');

jest.mock('../../data/dao/load-top-service-summaries');
jest.mock('../../data/dao/load-service');
jest.mock('../../data/dao/search-cases');

const loadTopServiceSummaries: JestMockFn = (require('../../data/dao/load-top-service-summaries'): any)
  .default;
const loadService: JestMockFn = (require('../../data/dao/load-service'): any)
  .default;
const searchCases: JestMockFn = (require('../../data/dao/search-cases'): any)
  .default;

const MOCK_SERVICE_SUMMARIES: ServiceSummary[] = [
  {
    name: 'Cosmic Incursion',
    code: 'CSMCINC',
    description: 'Something is threatening the fabric of the universe',
    group: 'ultimate',
  },
];

const MOCK_SERVICE: Service = {
  name: 'Cosmic Incursion',
  description: 'Bad things getting in from other universes',
  code: 'CSMCINC',
  contactRequirement: 'REQUIRED',
  locationRequirement: 'VISIBLE',
  attributes: [
    {
      required: false,
      type: 'TEXT',
      code: 'ST-CMTS',
      description: 'Please provide any other relevant information:',
      values: null,
      conditionalValues: null,
      dependencies: null,
    },
    {
      required: false,
      type: 'STRING',
      code: 'INFO-CSIRMV1',
      description:
        '**All cosmic incursion cases should be followed up with a phone call to Alpha Flight.**',
      values: null,
      conditionalValues: null,
      dependencies: null,
    },
    {
      required: true,
      type: 'SINGLEVALUELIST',
      code: 'SR-CSIRMV1',
      description: 'How many dimensions have been breached?',
      values: [
        { key: 'One', name: 'One' },
        { key: 'Two', name: 'Two' },
        { key: 'Three', name: 'Three' },
        { key: 'More than Three', name: 'More than Three' },
      ],
      conditionalValues: [],
      dependencies: null,
    },
  ],
};

let store;

beforeEach(() => {
  store = getStore();
  store.liveAgentAvailable = true;
  store.ui.visibleWidth = 1300;
  searchCases.mockReturnValue(new Promise(() => {}));
});

describe('request form', () => {
  let data;

  beforeEach(async () => {
    loadTopServiceSummaries.mockReturnValue(MOCK_SERVICE_SUMMARIES);

    const ctx = makeServerContext('/request');
    data = (await RequestLayout.getInitialProps(ctx)).data;
  });

  test('getInitialProps', () => {
    expect(data.view).toEqual('home');
  });
});

describe('translate page', () => {
  let data;

  beforeEach(async () => {
    loadTopServiceSummaries.mockReturnValue(MOCK_SERVICE_SUMMARIES);

    const ctx = makeServerContext('/request', { translate: '1' });
    data = (await RequestLayout.getInitialProps(ctx)).data;
  });

  test('getInitialProps', () => {
    expect(data.view).toEqual('translate');
  });
});

describe('existing service page', () => {
  beforeEach(() => {
    loadService.mockReturnValue(MOCK_SERVICE);
  });

  test('getInitialProps', async () => {
    const ctx = makeServerContext('/request', { code: 'CSMCINC' });
    const data = (await RequestLayout.getInitialProps(ctx)).data;
    expect(data.view).toEqual('request');
  });

  test('rendering phone', async () => {
    const ctx = makeServerContext(
      '/request',
      { code: 'CSMCINC' },
      { isPhone: true }
    );
    const data = (await RequestLayout.getInitialProps(ctx)).data;
    const component = renderer.create(
      <RequestLayout data={data} store={store} />,
      { createNodeMock: () => document.createElement('div') }
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('rendering phone location step', async () => {
    const ctx = makeServerContext(
      '/request',
      { code: 'CSMCINC', stage: 'location' },
      { isPhone: true }
    );
    const data = (await RequestLayout.getInitialProps(ctx)).data;
    const component = renderer.create(
      <RequestLayout data={data} store={store} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('missing service page', () => {
  let ctx;
  let data;

  beforeEach(async () => {
    loadService.mockReturnValue(null);

    ctx = makeServerContext('/request', { code: 'CSMCINC' });
    data = (await RequestLayout.getInitialProps(ctx)).data;
  });

  test('getInitialProps', () => {
    // flow check
    if (ctx.res) {
      expect(ctx.res.statusCode).toEqual(404);
    } else {
      expect(ctx.res).toBeDefined();
    }

    switch (data.view) {
      case 'request':
        expect(data.props.service).toBeNull();
      default:
        expect(data.view).toEqual('request');
    }
  });

  test('rendering', () => {
    const component = renderer.create(
      <RequestLayout data={data} store={store} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('routeToServiceForm', () => {
  let requestLayout;

  beforeEach(() => {
    const props: any = {
      data: {},
      store,
    };
    requestLayout = new RequestLayout(props);
  });

  it('defaults to questions', () => {
    requestLayout.routeToServiceForm('CSMCINC');
    expect(Router.push).toHaveBeenCalledWith(
      '/request?code=CSMCINC',
      '/request/CSMCINC'
    );
  });

  it('includes the stage', () => {
    requestLayout.routeToServiceForm('CSMCINC', 'contact');
    expect(Router.push).toHaveBeenCalledWith(
      '/request?code=CSMCINC&stage=contact',
      '/request/CSMCINC/contact'
    );
  });
});
