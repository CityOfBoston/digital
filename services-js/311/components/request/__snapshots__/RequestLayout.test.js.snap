// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`existing service page rendering phone 1`] = `
<div>
  <nav
    className="nv-s nv-s--y nv-s--sticky"
  >
    <input
      aria-hidden={true}
      className="nv-s-tr"
      id="nv-s-tr"
      type="checkbox"
    />
    <ul
      className="nv-s-l"
    >
      <li
        className="nv-s-l-i"
      >
        <label
          className="nv-s-l-b"
          htmlFor="nv-s-tr"
        >
          Navigation
        </label>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a nv-s-l-a--active"
          href="/"
          onClick={[Function]}
        >
          Make a request
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/search"
          onClick={[Function]}
        >
          Search cases
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/services"
          onClick={[Function]}
        >
          All services
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/faq"
          onClick={[Function]}
        >
          FAQ
        </a>
      </li>
    </ul>
  </nav>
  <div
    className={
      Object {
        "data-css-1d1pmz0": "",
      }
    }
    role="main"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-pmaldw": "",
        }
      }
    />
    <div
      className={
        Object {
          "data-css-1prrpm3": "",
        }
      }
    >
      <div
        className="p-a300 p-a800--xl br br-t400 br--y css-1aypqv0"
        style={Object {}}
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="sh sh--y"
          >
            <h1
              className="sh-title"
            >
              Cosmic Incursion
            </h1>
          </div>
          <div
            className="t--intro m-v300"
          >
            Bad things getting in from other universes
          </div>
          <div
            className="g g--top"
          >
            <div
              className="g--7 m-v500"
            >
              <div
                className="txt"
              >
                <label
                  className="txt-l"
                  htmlFor="QuestionsPane-description"
                  style={
                    Object {
                      "marginTop": 0,
                    }
                  }
                >
                  Description
                </label>
                <textarea
                  className="txt-f"
                  id="QuestionsPane-description"
                  name="description"
                  onChange={[Function]}
                  placeholder="Please describe your request"
                  rows="5"
                  value=""
                />
              </div>
              <div>
                <div
                  className="txt"
                >
                  <label
                    className="txt-l"
                    htmlFor="ST-CMTS"
                  >
                    Please provide any other relevant information:
                     
                  </label>
                  <textarea
                    aria-required={false}
                    className="txt-f"
                    id="ST-CMTS"
                    name="ST-CMTS"
                    onChange={[Function]}
                    rows="5"
                    value=""
                  />
                </div>
              </div>
              <hr
                aria-hidden={true}
                className="hr hr--dash m-v500"
              />
              <div>
                <div
                  className="txt"
                >
                  <label
                    className="txt-l"
                    htmlFor="INFO-CSIRMV1"
                  >
                    **All cosmic incursion cases should be followed up with a phone call to Alpha Flight.**
                     
                  </label>
                  <input
                    aria-required={false}
                    className="txt-f"
                    id="INFO-CSIRMV1"
                    name="INFO-CSIRMV1"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <hr
                aria-hidden={true}
                className="hr hr--dash m-v500"
              />
              <div>
                <div
                  aria-labelledby="SR-CSIRMV1-label"
                  role="radiogroup"
                >
                  <div
                    className="m-v300"
                  >
                    <label
                      className="txt-l"
                      id="SR-CSIRMV1-label"
                    >
                      How many dimensions have been breached?
                       
                      <span
                        className="t--req"
                      >
                        Required
                      </span>
                    </label>
                  </div>
                  <div
                    className="g"
                  >
                    <div
                      className="g--12"
                    >
                      <div
                        className="ra"
                      >
                        <input
                          checked={false}
                          className="ra-f"
                          id="SR-CSIRMV1-One"
                          name="SR-CSIRMV1"
                          onChange={[Function]}
                          type="radio"
                          value="One"
                        />
                        <label
                          className="ra-l"
                          htmlFor="SR-CSIRMV1-One"
                        >
                          One
                        </label>
                      </div>
                      <div
                        className="ra"
                      >
                        <input
                          checked={false}
                          className="ra-f"
                          id="SR-CSIRMV1-Two"
                          name="SR-CSIRMV1"
                          onChange={[Function]}
                          type="radio"
                          value="Two"
                        />
                        <label
                          className="ra-l"
                          htmlFor="SR-CSIRMV1-Two"
                        >
                          Two
                        </label>
                      </div>
                      <div
                        className="ra"
                      >
                        <input
                          checked={false}
                          className="ra-f"
                          id="SR-CSIRMV1-Three"
                          name="SR-CSIRMV1"
                          onChange={[Function]}
                          type="radio"
                          value="Three"
                        />
                        <label
                          className="ra-l"
                          htmlFor="SR-CSIRMV1-Three"
                        >
                          Three
                        </label>
                      </div>
                      <div
                        className="ra"
                      >
                        <input
                          checked={false}
                          className="ra-f"
                          id="SR-CSIRMV1-More than Three"
                          name="SR-CSIRMV1"
                          onChange={[Function]}
                          type="radio"
                          value="More than Three"
                        />
                        <label
                          className="ra-l"
                          htmlFor="SR-CSIRMV1-More than Three"
                        >
                          More than Three
                        </label>
                      </div>
                      <span />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="g--5 m-v500"
            >
              <div
                className="br br-a200"
              >
                <div
                  className="css-19ub58"
                  onClick={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                  onDragOver={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  style={Object {}}
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    height="324"
                    src="/assets/img/image-upload.png"
                    style={
                      Object {
                        "display": "block",
                        "height": "auto",
                        "width": "100%",
                      }
                    }
                    width="479"
                  />
                  <input
                    accept="image/*"
                    multiple={false}
                    onChange={[Function]}
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="file"
                  />
                </div>
                <div
                  className="br br-t200"
                  style={
                    Object {
                      "position": "relative",
                    }
                  }
                >
                  <button
                    className="btn btn--w btn--b"
                    onClick={[Function]}
                    style={
                      Object {
                        "visibility": "visible",
                        "width": "100%",
                      }
                    }
                    type="button"
                  >
                    Upload Photo
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="g m-v500 css-1k8tu3j"
          >
            <div
              className="g--9 t--info m-v200 css-hn0yet"
            >
              <span>
                Please fill out 
                <span
                  className="t--req"
                >
                  required
                </span>
                 fields to continue
              </span>
            </div>
            <button
              className="btn g--3"
              disabled={true}
              type="submit"
            >
              Next
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`existing service page rendering phone location step 1`] = `
<div>
  <nav
    className="nv-s nv-s--y nv-s--sticky"
  >
    <input
      aria-hidden={true}
      className="nv-s-tr"
      id="nv-s-tr"
      type="checkbox"
    />
    <ul
      className="nv-s-l"
    >
      <li
        className="nv-s-l-i"
      >
        <label
          className="nv-s-l-b"
          htmlFor="nv-s-tr"
        >
          Navigation
        </label>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a nv-s-l-a--active"
          href="/"
          onClick={[Function]}
        >
          Make a request
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/search"
          onClick={[Function]}
        >
          Search cases
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/services"
          onClick={[Function]}
        >
          All services
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/faq"
          onClick={[Function]}
        >
          FAQ
        </a>
      </li>
    </ul>
  </nav>
  <div
    className={
      Object {
        "data-css-1d1pmz0": "",
      }
    }
    role="main"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-pmaldw": "",
        }
      }
    />
    <div
      className={
        Object {
          "data-css-1msyv6a": "",
        }
      }
    >
      <div
        className=" br br-t400 br--y css-1aypqv0"
        style={Object {}}
      >
        <div
          className={
            Object {
              "data-css-1wu02iq": "",
            }
          }
        >
          <div
            className="p-a300 css-jvj6qy"
          >
            <form
              action="#"
              className="sf sf--sm sf--y"
              onSubmit={[Function]}
            >
              <div
                className="sf-i"
              >
                <input
                  aria-label="Address search field"
                  className="sf-i-f"
                  onChange={[Function]}
                  placeholder="Search for a street address or intersection…"
                  type="text"
                  value=""
                />
                <button
                  className="sf-i-b"
                  disabled={true}
                  type="submit"
                >
                  Search
                </button>
              </div>
            </form>
            <div
              className={
                Object {
                  "data-css-129qykh": "",
                }
              }
            >
              <div
                className="m-v400"
              />
            </div>
            <div
              className="g css-1k8tu3j"
            >
              <div
                className="g--7 t--subinfo m-v200"
              >
                <a
                  href="javascript:void(0)"
                  onClick={[Function]}
                >
                  Continue without location
                </a>
              </div>
              <button
                className="btn g--5"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Next
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`missing service page rendering 1`] = `
<div>
  <nav
    className="nv-s nv-s--y nv-s--sticky"
  >
    <input
      aria-hidden={true}
      className="nv-s-tr"
      id="nv-s-tr"
      type="checkbox"
    />
    <ul
      className="nv-s-l"
    >
      <li
        className="nv-s-l-i"
      >
        <label
          className="nv-s-l-b"
          htmlFor="nv-s-tr"
        >
          Navigation
        </label>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a nv-s-l-a--active"
          href="/"
          onClick={[Function]}
        >
          Make a request
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/search"
          onClick={[Function]}
        >
          Search cases
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/services"
          onClick={[Function]}
        >
          All services
        </a>
      </li>
      <li
        className="nv-s-l-i"
      >
        <a
          className="nv-s-l-a "
          href="/faq"
          onClick={[Function]}
        >
          FAQ
        </a>
      </li>
    </ul>
  </nav>
  <div
    className={
      Object {
        "data-css-1d1pmz0": "",
      }
    }
    role="main"
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-pmaldw": "",
        }
      }
    />
    <div
      className={
        Object {
          "data-css-1prrpm3": "",
        }
      }
    >
      <div
        className="p-a300 p-a800--xl br br-t400 br--y css-1aypqv0"
        style={Object {}}
      >
        <div
          className="sh sh--y"
        >
          <h1
            className="sh-title"
          >
            Service not found
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
